guide/src/class.md

```diff
@@ -1094,6 +1113,90 @@ enum BadSubclass {
 
 `#[pyclass]` enums are currently not interoperable with `IntEnum` in Python.
 
+### Complex enums
+
+An enum is complex if it has any non-unit (struct or tuple) variants.
+
+Currently PyO3 supports only struct variants in a complex enum. Support for unit and tuple variants is planned.
+
+PyO3 adds a class attribute for each variant, which may be used to construct values and in match patterns. PyO3 also provides getter methods for all fields of each variant.
+
+```rust
+# use pyo3::prelude::*;
+#[pyclass]
+enum Shape {
+    Circle { radius: f64 },
+    Rectangle { width: f64, height: f64 },
+    RegularPolygon { side_count: u32, radius: f64 },
+    Nothing { },
+}
+
+Python::with_gil(|py| {
+    let def_count_vertices = if py.version_info() >= (3, 10) { r#"
+        def count_vertices(cls, shape):
+            match shape:
+                case cls.Circle():
+                    return 0
+                case cls.Rectangle():
+                    return 4
+                case cls.RegularPolygon(side_count=n):
+                    return n
+                case cls.Nothing():
+                    return 0
+    "# } else { r#"
+        def count_vertices(cls, shape):
+            if isinstance(shape, cls.Circle):
+                return 0
+            elif isinstance(shape, cls.Rectangle):
+                return 4
+            elif isinstance(shape, cls.RegularPolygon):
+                n = shape.side_count
+                return n
+            elif isinstance(shape, cls.Nothing):
+                return 0
+    "# };
```

Comment:
This solves the version issue, but it makes the example ugly...
A more readable but less correct alternative would be to simply skip this test on Python < 3.10.

Reply:
You could `!#[cfg(Py_3_10)]` at the top of the example, it of sight of users even.

Given I've just set to merge, let's clean this up in a follow up PR.